### PR TITLE
makefile:add initial documentation and clean up aliases

### PR DESCRIPTION
**Pull request by avisiedo**
- Add basic usage of the current rules.
- Explain and break down the set of makefile parts..
- More detailed information about specific makefile parts.
- Clean-up alias rules.

Signed-off-by: Alejandro Visiedo <avisiedo@redhat.com>

### CODE REVIEW

File: mk/README.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+- `mk/projectdir.mk`: It uses to be the first include and just store
+  at the `PROJECT_DIR` variable the path to the repository in the
+  file system.
+- `mk/private.mk`: Not used into this project, but it could define
+  by using conditional assignment values that we want to set for
+  our make rules; useful for the specific rules related with our
+  tech-stack, or when building container images, to customize
+  our own registry (quay.io/my-username); provide a mechanism
+  of customization to avoid repeat steps manually as `export MY_VAR=my-value`.
```
Comment (jlsherrill) [inline mk/README.md:98]: I think we should either remove the text about private.mk or add support for it in in our project (likely adding it to includes.mk, adding a blank one, and adding it to git ignore?)

File: mk/README.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+@Miscelanea
+
+- `mk/printvars.mk`: It helps to fix wrong behaviors so it print out
+  the variable name and the value not evaluated, so we can realize
+  what final values could be applied and how should the values be
+  expanded.
+- `mk/alias.mk`: It provides compatibility with the initial set of
+  rules in the Makefile, so that it would avoid impact during a
+  transition period to the new set of rules.
```
Comment (jlsherrill) [inline mk/README.md:146]: our project is so new, i think we should just drop the aliases alltogether, but i'll bring this up in our next standup :)

**Comment by jlsherrill** [comment]
overall a BIG +1, just a few small comments

File: mk/README.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+├── docker.mk
+├── go-rules.mk
+├── help.mk                 # Rule to print out generated help content from the own Makefile files
+├── includes.mk             # Entry point included into the main Makefile
+├── meta-alias.mk           # Group title included from the mk/includes.mk file
+├── meta-db.mk
+├── meta-docker.mk
+├── meta-general.mk
+├── plantuml.mk
+├── printvars.mk            # print the list of defined variables, without evaluate the value
```
[the remark this answers is not in the document]
Comment (avisiedo) [reply mk/README.md:24]: thanks for this; this needs some clarification; I have checked out, and not always is printing out the evaluated value. Only when the last assignment was using the `:=` and `?=` operators, else it prints out the expression, instead the evaluated value.

Example:

For this `makefile` file included which define some variables in different ways:

```makefile
EXAMPLE_01 := example_01
EXAMPLE_02 = example_02
EXAMPLE_03 += example_03
EXAMPLE_04 ?= example_04

EXAMPLE_05 = $(EXAMPLE_01) example_05
EXAMPLE_06 = $(EXAMPLE_02) example_06
EXAMPLE_07 = $(EXAMPLE_03) example_07
EXAMPLE_08 = $(EXAMPLE_04) example_08

EXAMPLE_09 += $(EXAMPLE_01) example_09
EXAMPLE_10 += $(EXAMPLE_02) example_10
EXAMPLE_11 += $(EXAMPLE_03) example_11
EXAMPLE_12 += $(EXAMPLE_04) example_12

EXAMPLE_01 := $(EXAMPLE_01)
EXAMPLE_02 = $(EXAMPLE_02)
EXAMPLE_03 += $(EXAMPLE_03)
EXAMPLE_04 ?= $(EXAMPLE_04)

```

The resulting output for `make printvars`

```raw
EXAMPLE_01=example_01
EXAMPLE_02=$(EXAMPLE_02)
EXAMPLE_03=example_03 $(EXAMPLE_03)
EXAMPLE_04=example_04
EXAMPLE_05=$(EXAMPLE_01) example_05
EXAMPLE_06=$(EXAMPLE_02) example_06
EXAMPLE_07=$(EXAMPLE_03) example_07
EXAMPLE_08=$(EXAMPLE_04) example_08
EXAMPLE_09=$(EXAMPLE_01) example_09
EXAMPLE_10=$(EXAMPLE_02) example_10
EXAMPLE_11=$(EXAMPLE_03) example_11
EXAMPLE_12=$(EXAMPLE_04) example_12
```

[the remark this answers is not in the document]
Comment (avisiedo) [reply mk/README.md:24]: I think is better just to say `print out the variable values`, if that values are evaluated or not, depends on which operator was used the last and if it is including into the value some substitution or calling some command inside the variable value.

File: mk/README.md (hunk 1 above)
Comment (avisiedo) [reply mk/README.md]: Thanks! Updated.

File: mk/README.md (hunk 2 above)
Comment (avisiedo) [reply mk/README.md:140]: thanks! it does not make many sense the `mk/alias.mk` then, but keeping it until the next standup.

File: mk/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (avisiedo) [reply mk/README.md]: slight update into the comment about 'printvars.mk' file.

**Comment by jlsherrill** [comment]
Probably need to update this line:  https://github.com/content-services/content-sources-backend/blob/main/.github/workflows/content-sources-actions.yml

File: mk/README.md (hunk 1 above)
Comment (avisiedo) [reply mk/README.md]: thanks! I removed the references to `private.mk`; I didn't pay attention to the other parts.

File: mk/README.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
+# Makefile composition
+
+The repository is using a way to compose the `Makefile` file which
+empower the single responsability principal taken into the Makefile
+file usage.
+
+Very quickly, we have the following files:
```
Comment (swadeley) [inline mk/README.md:7]:
```suggestion
We have the following files:
```
Try not to make judgements about things (brief, easy, quick etc.)

File: mk/README.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+```raw
+Makefile                    # Minimal main Makefile
+mk
+├── README.md               # This documentation file
+├── db.mk
+├── docker.mk
+├── go-rules.mk
+├── help.mk                 # Rule to print out generated help content from the own Makefile files
```
Comment (swadeley) [inline mk/README.md:16]:
```suggestion
├── help.mk                 # Rule to print out generated help content from the Makefile's files
```

File: mk/README.md [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+├── plantuml.mk
+├── printvars.mk            # print out variables
+├── projectdir.mk           # Store at PROJECT_DIR the base directory for the repository
+├── swag.mk
+└── variables.mk            # Default values to the project variables that has not been
+                            # overrided by the environment variables nor by configs/config.yaml
+                            # file
+```
+
+## Quickly usage
```
Comment (swadeley) [inline mk/README.md:30]:
```suggestion
## Usage
```

File: mk/README.md [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+  ```sh
+  make docker-build docker-push DOCKER_IMAGE_BASE="quay.io/my-user/my-image"
+  ```
+
+  > See `make printvars` to check related variables.
+
+## More detailed explanation
+
+It uses single responsability principal and split basic or logical groupable
+rules in a single files.
```
Comment (swadeley) [inline mk/README.md:71]:
```suggestion
rules in a single file.
```

File: mk/README.md [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+- `Makefile`: Entrypoint and the only responsability is to include
+  the list of includes and set the dafault rule.
+- `mk/includes.mk`: This define the composition for your repository
+  and it reference other `mk/*` files, and the `mk/meta-*` files
+  which allow the `help` rule to print out the group title in the
+  right position.
+
+@Variables
+
+- `mk/projectdir.mk`: It uses to be the first include and just store
```
Comment (swadeley) [inline mk/README.md:86]:
```suggestion
- `mk/projectdir.mk`: It uses the first include and stores
```

File: mk/README.md [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+  at the `PROJECT_DIR` variable the path to the repository in the
+  file system.
+- `mk/variables.mk`: It defines the default values for variables.
+
+@Help infrastructure
+
+- `mk/meta-*.mk`: This files is just one line content each one,
```
Comment (swadeley) [inline mk/README.md:93]:
```suggestion
- `mk/meta-*.mk`: This file is just one line of content for each one,
```

File: mk/README.md [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+  but that allow us to print the group label into the help generated
+  as we want. It can not be written directly into the `mk/includes.mk`
+  file because of the order the files are processed when the `help`
+  rule is invoked.
+- `mk/help.mk`: Contain the rule that generate the help content.
```
Comment (swadeley) [inline mk/README.md:98]:
```suggestion
- `mk/help.mk`: Contains the rule that generates the help content.
```

File: mk/README.md [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+
+@Rules
+
+- `mk/go-rules.mk`: Provide useful rules to build/clean/test
```
Comment (swadeley) [inline mk/README.md:102]:
```suggestion
- `mk/go-rules.mk`: Provide useful rules to build, clean, and test
```
No slash in text please

File: mk/README.md [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+@Miscelanea
+
+- `mk/printvars.mk`: It helps to fix wrong behaviors so it print out
+  the variable name and the value not evaluated, so we can realize
+  what final values could be applied and how should the values be
+  expanded.
+
+## Contributing
+
+Do you have a rule or some set of rules that could be useful here? please
```
Comment (swadeley) [inline mk/README.md:134]:
```suggestion
Do you have a rule or some set of rules that could be useful here?
```
No need for "please" in techincal docs.

File: mk/README.md [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+do not hesitate to contribute to add and review them together. It does
```
Comment (swadeley) [inline mk/README.md:135]:
```suggestion
Do not hesitate to contribute and review them together. It does
```

**Comment by avisiedo** [comment]
@swadeley @jlsherrill thanks for the review! Updated!

**Review by jlsherrill — APPROVED** [review]

**Comment by jlsherrill** [comment]
Go ahead and merge, as this was an update to a previous pr.  Future Prs needs JIra tickets :)

**Comment by avisiedo** [comment]
@swadeley @jlsherrill thanks for the review and comments

**Pull request merged**